lib/widget/category_list.dart

```diff
@@ -12,34 +13,43 @@ class CategoryList extends StatelessWidget {
 
   final List<Category> categories;
 
-  Wrap _getCategoriesWidgets(List<Category> categories, BuildContext context) {
+  Wrap _getCategoriesWidgets(
+    List<Category> categories,
+    CategoryViewModel model,
+    BuildContext context,
+  ) {
     final List<Widget> list = <Widget>[];
     for (int i = 0; i < categories.length; i++) {
       if (categories[i].name != 'custom') {
         list.add(
           GestureDetector(
             onTap: () {
-              for (int y = 0; y < categories.length; y++) {
-                // vm.database.categoryDao
-                //     .updateCategory(categories[y].copyWith(focused: false));
-              }
+              // for (int y = 0; y < categories.length; y++) {
+              //   // vm.database.categoryDao
+              //   //     .updateCategory(categories[y].copyWith(focused: false));
+              // }
               // vm.database.categoryDao.updateCategory(
               //     categories[i].copyWith(focused: !categories[i].focused));
+              model.updateCategory(
+                  category: categories[i].name.toString(),
+                  id: categories[i].id.toString());
```

Comment:
It should be a string in the first place, which I think it is so no need to convert it again to be a string.